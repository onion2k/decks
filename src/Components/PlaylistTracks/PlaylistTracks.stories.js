import React from "react";
import { storiesOf } from "@storybook/react";
import { action } from "@storybook/addon-actions";
import PlaylistTracks from "./index.js";

const tracks = [];

storiesOf("PlaylistTracks", module).add("Basic", () => (
  <PlaylistTracks playlist={tracks} />
));
