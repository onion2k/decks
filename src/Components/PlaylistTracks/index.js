import React, { Component } from "react";

class PlaylistTracks extends Component {
  timeFormat(time) {
    if (!time) {
      return "";
    }

    var hrs = ~~(time / 3600);
    var mins = ~~((time % 3600) / 60);
    var secs = time % 60;

    var ret = "";

    if (hrs > 0) {
      ret += "" + hrs + ":" + (mins < 10 ? "0" : "");
    }

    ret += "" + mins + ":" + (secs < 10 ? "0" : "");
    ret += "" + Math.round(secs);
    return ret;
  }
  render() {
    let tracklist;

    if (this.props.playlist.length > 0) {
      tracklist = this.props.playlist.map(track => {
        return (
          <div
            key={track.videoId}
            className={
              "track" +
              (track.videoId === this.props.playing ? " playing" : "") +
              (track.found ? " found" : "")
            }
            onClick={e => this.props.onClick(track.videoId)}
          >
            <i
              className="fa fa-trash"
              aria-hidden="true"
              onClick={e => {
                e.stopPropagation();
                this.props.onDelete(track.videoId);
              }}
            />{" "}
            <i className="fa fa-music" aria-hidden="true" />{" "}
            {track.title || track.videoId} <span>{this.timeFormat(track.duration)}</span>
          </div>
        );
      });
    } else {
      tracklist = <div className="track">No tracks found</div>;
    }

    return <div className="Playlist">{tracklist}</div>;
  }
}

export default PlaylistTracks;
