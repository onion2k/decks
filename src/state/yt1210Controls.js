import { configure, decorate, observable, action } from "mobx";

import fx from "./yt1210Sounds";

configure({ enforceActions: true });

export default class yt1210Controls {

    constructor(pm){
      this.pm = pm;
    }
    
    videoId = null;
  
    playing = false;
    repeat = false;
    shuffle = false;
    crackle = true;
    video = true;
  
    // play track
    play = (id = null) => {
      if (id!==null) {
        this.videoId = this.pm.getCurrentTrack().videoId;
      } else {
        this.videoId = this.pm.getTrackById(id).videoId;
      }
      this.playing = true;
    };
  
    // pause track
    pause = () => {
      this.playing = !this.playing;
    };
  
    // stop track
    stop = () => {
      this.playing = false;
    };
  
    // next track
    next = () => {
      this.videoId = this.pm.getNextTrack().videoId;
      this.playing = true;
    };
  
    // copy playlist from youtube

    toggle_repeat = () => {
      this.repeat = !this.repeat;
      if (this.repeat) { fx.ogg_scratchin.play(); }
    }
  
    toggle_shuffle = () => {
      this.shuffle = !this.shuffle;
    };
  
    toggle_crackle = () => {
      this.crackle = !this.crackle;
    };

    toggle_video = () => {
      this.video = !this.video;
    };
  }
  
  decorate(yt1210Controls, {
    play: action,
    pause: action,
    stop: action,
    next: action,
    toggle_repeat: action,
    toggle_shuffle: action,
    toggle_crackle: action,
    toggle_video: action,
    videoId: observable,
    playing: observable,
    crackle: observable,
    video: observable,
    repeat: observable,
    shuffle: observable
  });
  
