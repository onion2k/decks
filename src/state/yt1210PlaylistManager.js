import { configure, decorate, observable, action, computed } from "mobx";

configure({ enforceActions: true });

export default class yt12010PlaylistManager {

    // Playlists

    playlist = 1;
    track = 0;
    currentTrack = { videoId: "", label: "" };

    playlists = [
        { "id": 1, "title": "Playlist 1", tracks: [
          { title: "", videoId: "BuVJEn9wk9Y", duration: "" },
          { title: "", videoId: "Rs38lKxmtI4", duration: "" },
          { title: "", videoId: "v_yTphvyiPU", duration: "" },
          { title: "", videoId: "e4Ao-iNPPUc", duration: "" },
          { title: "", videoId: "FALYmqt-7TQ", duration: "" },
          { title: "", videoId: "a3Z4RWZa9WA", duration: "" }
        ]},
        { "id": 2, "title": "Playlist 2", tracks: [
            { title: "", videoId: "a3Z4RWZa9WA", duration: "" },
            { title: "", videoId: "BuVJEn9wk9Y", duration: "" },
            { title: "", videoId: "Rs38lKxmtI4", duration: "" },
            { title: "", videoId: "v_yTphvyiPU", duration: "" },
            { title: "", videoId: "e4Ao-iNPPUc", duration: "" },
            { title: "", videoId: "FALYmqt-7TQ", duration: "" }
          ]}
    ];

    load = (id) => {
        this.playlist = id;
        // const pl = this.playlists.find((playlist) => playlist.id === this.playlist).tracks;
    };

    // add playlist
    playlistAdd = (title)=>{
        this.playlists.push({ id: this.playlists.length+1, title: title, tracks: [] });
    };

    // delete playlist
    playlistDelete = (index)=>{
        this.playlists.splice(index, 1);
    };

    // add track
    playlistAddTrack = (id, videoId)=>{
        this.playlist = this.playlists.find((playlist) => playlist.id === id);
        this.playlist.tracks.push({ title: "", videoId: videoId, duration: "" });
    };

    // delete track
    playlistDeleteTrack = (id, index)=>{
        this.playlist = this.playlists.find((playlist) => playlist.id === id);
        this.playlist.tracks.splice(index, 1);
    };

    get tracks() {
        return this.playlists.find((playlist) => playlist.id === this.playlist).tracks;
    };

    updateTrackData = (data, duration) => {
        const track = this.tracks.find((track)=>track.videoId === data.video_id);
              track.title = data.title;
              track.duration = duration;
        this.currentTrack = track;
    };

    getTracks = () => {
        return this.tracks;
    };

    getCurrentTrack = () => {
        return this.tracks[this.track];
    };

    getNextTrack = () => {
        return this.tracks[this.track++];
    };

    getTrackById = (id) => {
        this.track = this.tracks.findIndex((track) => track.videoId === id);
        return this.tracks[this.track];
    };
}

decorate(yt12010PlaylistManager, {
    currentTrack: observable,
    playlists: observable,
    playlist: observable,
    track: observable,
    tracks: computed,
    load: action,
    playlistAdd: action,
    playlistDelete: action,
    playlistAddTrack: action,
    playlistDeleteTrack: action,
    updateTrackData: action,
    getCurrentTrack: action,
    getNextTrack: action
});
