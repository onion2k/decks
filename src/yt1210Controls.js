// play
// stop
// pause
// next
// previous

// tonearm move
