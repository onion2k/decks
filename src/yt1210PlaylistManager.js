import { configure, decorate, observable, action, autorun } from "mobx";

export default class yt12010PlaylistManager {

    // Playlists

    playlist = [];

    playlists = [
        { "id": 1, "title": "Playlist 1", tracks: [
          { title: "", videoId: "BuVJEn9wk9Y", duration: "" },
          { title: "", videoId: "Rs38lKxmtI4", duration: "" },
          { title: "", videoId: "v_yTphvyiPU", duration: "" },
          { title: "", videoId: "e4Ao-iNPPUc", duration: "" },
          { title: "", videoId: "FALYmqt-7TQ", duration: "" },
          { title: "", videoId: "a3Z4RWZa9WA", duration: "" }
        ]},
        { "id": 2, "title": "Playlist 2", tracks: [0,1,2,3,4,5] },
        { "id": 3, "title": "Playlist 3", tracks: [0,1,2,3,4,5] },
        { "id": 4, "title": "Playlist 4", tracks: [0,1,2,3,4,5] }
    ];

    load = action((id) => {
        this.playlist = this.playlists.find((playlist) => playlist.id === id).tracks;
        this.videoId = this.playlist[0].videoId;
        this.playing = true;
    });

    // add playlist
    playlistAdd = action((title)=>{
        this.playlists.push({ id: this.playlists.length+1, title: title, tracks: [] });
    });

    // delete playlist
    playlistDelete = action((index)=>{
        this.playlists.splice(index, 1);
    });

    // add track
    playlistAddTrack = action((id, videoId)=>{
        this.playlist = this.playlists.find((playlist) => playlist.id === id);
        this.playlist.tracks.push({ title: "", videoId: videoId, duration: "" });
    });

    // delete track
    playlistDeleteTrack = action((id, index)=>{
        this.playlist = this.playlists.find((playlist) => playlist.id === id);
        this.playlist.tracks.splice(index, 1);
    });

    updateTrackData = action((data, duration) => {

    });

}

decorate(yt12010PlaylistManager, {
    playlist: observable,
    playlists: observable,
    load: action,
    playlistAdd: action,
    playlistDelete: action,
    playlistAddTrack: action,
    playlistDeleteTrack: action,
    updateTrackData: action
});