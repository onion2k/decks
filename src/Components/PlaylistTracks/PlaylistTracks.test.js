import React from "react";
import renderer from "react-test-renderer";
import { MemoryRouter } from 'react-router-dom'
import PlaylistTracks from ".";
import localStorage from 'jest-localstorage-mock';

import yt1210PlaylistManager from "../../state/yt1210PlaylistManager"
const playlistManagerState = new yt1210PlaylistManager();

import yt1210Controls from "../../state/yt1210Controls"
const controlState = new yt1210Controls();

describe("Playlist Tracks", () => {
  const onClick = jest.fn();

  it("renders properly", () => {
    const tracks = renderer
      .create(
        <MemoryRouter>
          <PlaylistTracks playlistManager={playlistManagerState} playlistControls={controlState} />
        </MemoryRouter>
      )
      .toJSON();
    expect(tracks).toMatchSnapshot();
  });
});
