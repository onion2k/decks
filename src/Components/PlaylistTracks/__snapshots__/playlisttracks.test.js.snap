// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Playlist Tracks renders properly 1`] = `
<div
  className="Playlist"
>
  <div
    className="track"
    onClick={[Function]}
  >
    <i
      aria-hidden="true"
      className="fa fa-trash"
      onClick={[Function]}
    />
     
    <i
      aria-hidden="true"
      className="fa fa-music"
    />
     
    BuVJEn9wk9Y
     
    <span>
      
    </span>
  </div>
  <div
    className="track"
    onClick={[Function]}
  >
    <i
      aria-hidden="true"
      className="fa fa-trash"
      onClick={[Function]}
    />
     
    <i
      aria-hidden="true"
      className="fa fa-music"
    />
     
    Rs38lKxmtI4
     
    <span>
      
    </span>
  </div>
  <div
    className="track"
    onClick={[Function]}
  >
    <i
      aria-hidden="true"
      className="fa fa-trash"
      onClick={[Function]}
    />
     
    <i
      aria-hidden="true"
      className="fa fa-music"
    />
     
    v_yTphvyiPU
     
    <span>
      
    </span>
  </div>
  <div
    className="track"
    onClick={[Function]}
  >
    <i
      aria-hidden="true"
      className="fa fa-trash"
      onClick={[Function]}
    />
     
    <i
      aria-hidden="true"
      className="fa fa-music"
    />
     
    e4Ao-iNPPUc
     
    <span>
      
    </span>
  </div>
  <div
    className="track"
    onClick={[Function]}
  >
    <i
      aria-hidden="true"
      className="fa fa-trash"
      onClick={[Function]}
    />
     
    <i
      aria-hidden="true"
      className="fa fa-music"
    />
     
    FALYmqt-7TQ
     
    <span>
      
    </span>
  </div>
  <div
    className="track"
    onClick={[Function]}
  >
    <i
      aria-hidden="true"
      className="fa fa-trash"
      onClick={[Function]}
    />
     
    <i
      aria-hidden="true"
      className="fa fa-music"
    />
     
    a3Z4RWZa9WA
     
    <span>
      
    </span>
  </div>
</div>
`;
